### PR TITLE
Implement init command

### PR DESCRIPTION
**Pull request by mkrisher**
This PR implements the `init` command.

```
$ cr -- init --dryrun
    Finished dev [unoptimized + debuginfo] target(s) in 0.10s
     Running `target/debug/tembo init --dryrun`
config file would be created at: /Users/mkrisher/.config/tembo/configuration.toml
```
```
$ cr -- init --dryrun --file-path .
    Finished dev [unoptimized + debuginfo] target(s) in 0.04s
     Running `target/debug/tembo init --dryrun --file-path .`
config file would be created at: /Users/mkrisher/projects/tembo-cli/./.config/tembo/configuration.toml
```

It does a few things, first it accepts two optional parameters, dryrun and file-path. dryrun being set means no file is written. file-path when provided allows the user to specify the directory to write the config file into. This means users can have a global config written in their user directory and others in say an individual project directory. future commands will need to support a file-path flag.

Simply running `cargo run -- init` will create a configuration file in the $HOME/.config/tembo.

If an existing configuration file exists, it will not be overwritten.

Right now the file simply contains `[configuration]`. We'll add to it later.

Unit tests are included. The integration test received a very small change. More integration tests to come as we add commands.

### CODE REVIEW

File: .github/workflows/cli.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
       - name: Install minimal nightly with clippy and rustfmt
         uses: moonrepo/setup-rust@v0
         with:
-          channel: stable
+          channel: nightly
```
Comment (mkrisher) [inline .github/workflows/cli.yml:26]: annotation: switches to the nightly toolchain

File: Cargo.toml [hunk 2]
```diff
@@ -17,6 +17,7 @@ clap = { version = "4.3.12", features = ["cargo", "wrap_help"] }
 clap_complete = "4.3.2"
 serde_json = "1.0.91"
 serde_yaml = "0.9.17"
+home = "0.5.5"
```
Comment (mkrisher) [inline Cargo.toml:20]: annotation: used to determine the home directory

File: rust-toolchain [hunk 3]
```diff
@@ -0,0 +1 @@
+nightly
```
Comment (mkrisher) [inline rust-toolchain:1]: annotation: using nightly locally

File: src/cmd/init.rs [hunk 4]
```diff
@@ -1,23 +1,22 @@
...
 pub fn make_subcommand() -> Command {
     Command::new("init")
         .about("Initializes a local environment or project; generates configuration")
-        .arg(
-            Arg::new("dry-run")
-                .short('d')
-                .long("dry run")
-                .value_name("dry-run"),
-        )
+        .arg(arg!(-d --dryrun "Provides an input file to the program"))
```
Comment (mkrisher) [inline src/cmd/init.rs:13]: annotation: adds an optional dryrun flag, handy for determining where the config file would be written

File: src/cmd/init.rs [hunk 5]
```diff
@@ -1,23 +1,22 @@
...
         .arg(
```
Comment (mkrisher) [inline src/cmd/init.rs:14]: annotation: allows for a file_path param, so users can store a project specific configuration file in addition to the global maybe

File: src/cmd/init.rs [hunk 6]
```diff
@@ -27,11 +26,188 @@ pub fn make_subcommand() -> Command {
...
+
+    let mut path: PathBuf = get_dir(args);
+    path.push(".config");
+    path.push("tembo");
+    path.push(file_name);
+
+    path
+}
+
+fn get_dir(args: &ArgMatches) -> PathBuf {
```
Comment (mkrisher) [inline src/cmd/init.rs:60]: annotation: used to determine the directory to write the config file to, home dir is used as the default, both relative and absolute file-paths are supported

File: src/cmd/init.rs [hunk 7]
```diff
@@ -27,11 +26,188 @@ pub fn make_subcommand() -> Command {
...
+            p.to_path_buf()
+        }
+    } else {
+        // if file-path was not provided, use the home directory as a default
+        let home_dir = home::home_dir();
+
+        // if home directory can not be determined, use the current directory
+        match home_dir {
+            Some(path) => path,
+            None => env::current_dir().expect("Unable to determine the current directory"),
```
Comment (mkrisher) [inline src/cmd/init.rs:77]: annotation: if the home directory can't be safely determined, we try to use the current directory, if that can't be determined we'll throw an error

File: src/cmd/init.rs [hunk 8]
```diff
@@ -27,11 +26,188 @@ pub fn make_subcommand() -> Command {
...
+        }
+    }
+}
+
+fn create_path_and_write_config_file(path: &PathBuf) -> Result<(), Box<dyn Error>> {
```
Comment (mkrisher) [inline src/cmd/init.rs:82]: annotation: we have to create the directory before trying to write a file there

File: src/cmd/init.rs [hunk 9]
```diff
@@ -27,11 +26,188 @@ pub fn make_subcommand() -> Command {
...
+
+fn create_config_dir(dir_path: &str) -> Result<(), Box<dyn Error>> {
+    fs::create_dir_all(dir_path)?;
+    Ok(())
+}
 
-    // TODO: implement init command to generate a config file
-    println!("Soon this will persist a configuration file");
+fn create_config_file(path: &str) -> Result<(), Box<dyn Error>> {
+    let mut file = File::create_new(&path)?; // don't overwrite existing file at path
```
Comment (mkrisher) [inline src/cmd/init.rs:102]: annotation: using create_new so we don't overwrite an existing file, that would stink if a large configuration had already been built up

**Review by ChuckHend — APPROVED** [review]

**Pull request merged**